src/unk_020648EC.c

```diff
@@ -0,0 +1,78 @@
+#include "gymmick.h"
+#include "overlay_04.h"
+#include "save_misc_data.h"
+#include "scrcmd.h"
+#include "script.h"
+
+typedef void (*FieldSystemFunc1)(FieldSystem*);
+typedef BOOL (*FieldSystemFunc2)(FieldSystem*, u32, u32, u32, u32);
+
+static const FieldSystemFunc1 _020FE214[];
+static const FieldSystemFunc1 _020FE1EC[];
+static const FieldSystemFunc2 _020FE23C[];
+
+void sub_020648EC(FieldSystem* fsys) {
+    int gymmickType = SavGymmick_GetType(Sav2_GetGymmickPtr(Fsys_GetSaveDataPtr(fsys)));
+    if (gymmickType) {
+        _020FE214[gymmickType](fsys);
+    }
+}
+
+void sub_02064910(FieldSystem* fsys) {
+    int gymmickType = SavGymmick_GetType(Sav2_GetGymmickPtr(Fsys_GetSaveDataPtr(fsys)));
+    if (gymmickType) {
+        FieldSystemFunc1 func = _020FE1EC[gymmickType];
+        if (func) {
```

Comment:
`func` is not a boolean, so explicit comparison to `NULL` is desired for the sake of communication.

Reply:
Done